File: vv.py
import calendar
import datetime

now = datetime.datetime.now()
dotw = calendar.day_name[now.weekday()]

y = int(now.strftime("%Y"))
m = int(now.strftime("%m"))
d = int(now.strftime("%d"))

def amTimes(am_open, am_close):
  print(f"OPEN:")
  print(f"Open Swim (ALL AGES)")
  print(am_open.strftime("%I") + ":" + am_open.strftime("%M"), am_open.strftime("%p"), "-", am_close.strftime("%I") + ":" + am_close.strftime("%M"), am_close.strftime("%p"))

def familyNight(family_night_open, family_night_close):
  print("OPEN:")
  print("Family Night - $15 per family (ALL AGES)")
  print(family_night_open.strftime("%I") + ":" + family_night_open.strftime("%M"), family_night_open.strftime("%p"), "-", family_night_close.strftime("%I") + ":" + family_night_close.strftime("%M"), family_night_close.strftime("%p"))

def halfOff(half_off_open, half_off_close):
  print("OPEN:")
  print("Open Swim - Half Price! (ALL AGES)")
  print(half_off_open.strftime("%I") + ":" + half_off_open.strftime("%M"), half_off_open.strftime("%p"), "-", half_off_close.strftime("%I") + ":" + half_off_close.strftime("%M"), half_off_close.strftime("%p"))

if datetime.datetime(year=2023, month=6, day=2, hour=00, minute=00) <= now < datetime.datetime(year=2023, month=9, day=1, hour=00, minute=00):
    am_open = datetime.datetime(year=y, month=m, day=d, hour=12, minute=00)
    if dotw == "Saturday":
       am_close = datetime.datetime(year=y, month=m, day=d, hour=18, minute=00)
       if am_open <= now < am_close:
          amTimes(am_open, am_close)
       elif am_close <= now:
          print("CLOSED")
    else:
        am_close = datetime.datetime(year=y, month=m, day=d, hour=17, minute=30)
        adult_swim_open = datetime.datetime(year=y, month=m, day=d, hour=17, minute=30)
        adult_swim_close = datetime.datetime(year=y, month=m, day=d, hour=18, minute=30)
        pm_open = datetime.datetime(year=y, month=m, day=d, hour=18, minute=30)
        pm_close = datetime.datetime(year=y, month=m, day=d, hour=20, minute=30)
        if am_open <= now < am_close:
            amTimes(am_open, am_close)
        elif adult_swim_open <= now < adult_swim_close:
           print("OPEN:")
           print("Current Walking and Adult Lap Swim (18+ ONLY)")
           print(f'{adult_swim_open.strftime("%I")}:{adult_swim_open.strftime("%M")} {adult_swim_open.strftime("%p")} - {adult_swim_close.strftime("%I")}:{adult_swim_close.strftime("%M")} {adult_swim_close.strftime("%p")}')
        elif pm_open <= now < pm_close:
           if (dotw == "Wednesday") or (dotw == "Sunday"):
              familyNight(pm_open, pm_close)
           else:
              halfOff(pm_open, pm_close)
        elif pm_close <= now:
           print("CLOSED")

File: vv.js
var now = new Date();
const daysOfWeek = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
var dotw = daysOfWeek[now.getDay()];
var y = now.getFullYear();
var m = now.getMonth();
var d = now.getDate();

function amTimes(am_open, am_close) {
  document.getElementById("vv-head").innerHTML = "THE POOL IS CURRENTLY <u>OPEN</u>:";
  document.getElementById("vv-body").innerHTML = "<u>Open Swim (ALL AGES)</u><br><b>" + am_open.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + " - " + am_close.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + "</b>";
}

function familyNight(family_night_open, family_night_close) {
  document.getElementById("vv-head").innerHTML = "THE POOL IS CURRENTLY <u>OPEN</u>:";
  document.getElementById("vv-body").innerHTML = "<u>Family Night - $15 per family (ALL AGES)</u><br><b>" + family_night_open.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + " - " + family_night_close.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + "</b>";
}

function halfOff(half_off_open, half_off_close) {
  document.getElementById("vv-head").innerHTML = "OPEN:";
  document.getElementById("vv-body").innerHTML = "<u>Open Swim - Half Price! (ALL AGES)</u><br><b>" + half_off_open.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + " - " + half_off_close.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + "</b>";
}

function morning(am_open) {
  document.getElementById("vv-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
  document.getElementById("vv-body").innerHTML = "We will be open at " + am_open.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  });
}

var am_open = new Date(y, m, d, 12, 0);
if (dotw === "Saturday") {
  var am_close = new Date(y, m, d, 18, 0);
  if (am_open <= now && now < am_close) {
    amTimes(am_open, am_close);
  } else if (am_close <= now) {
    document.getElementById("vv-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
  } else if (now < am_open) {
    morning(am_open);
  }
} else {
  var am_close = new Date(y, m, d, 17, 30);
  var adult_swim_open = new Date(y, m, d, 17, 30);
  var adult_swim_close = new Date(y, m, d, 18, 30);
  var pm_open = new Date(y, m, d, 18, 30);
  var pm_close = new Date(y, m, d, 20, 30);
  if (am_open <= now && now < am_close) {
    amTimes(am_open, am_close);
  } else if (adult_swim_open <= now && now < adult_swim_close) {
    document.getElementById("vv-head").innerHTML = "THE POOL IS CURRENTLY <u>OPEN</u>:";
    document.getElementById("vv-body").innerHTML = "<u>Current Walking and Adult Lap Swim (18+ ONLY)</u><br><b>" + adult_swim_open.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + " - " + adult_swim_close.toLocaleString('en-US', {
    hour: 'numeric',
    minute: 'numeric',
    hour12: true
  }) + "</b>";
  } else if (pm_open <= now && now < pm_close) {
    if (dotw === "Wednesday" || dotw === "Sunday") {
      familyNight(pm_open, pm_close);
    } else {
      halfOff(pm_open, pm_close);
    }
  } else if (pm_close <= now) {
    document.getElementById("vv-head").innerHTML = "THE POOL IS CURRENTLY <u>CLOSED</u>";
  } else if (now < am_open) {
    morning(am_open);
  }
}